--- a/drivers/net/ethernet/allwinner/gmac/gmac_core.c
+++ b/drivers/net/ethernet/allwinner/gmac/gmac_core.c
@@ -217,6 +217,8 @@
 			|| phy_interface == PHY_INTERFACE_MODE_GMII)
 		priv_clk_reg |= 0x00000002;
 
+        priv_clk_reg |= (0x00000003<<10);
+
 	writel(priv_clk_reg, priv->gmac_clk_reg + GMAC_CLK_REG);
 }
 
@@ -1753,6 +1755,6 @@
 module_init(gmac_init);
 module_exit(gmac_remove);
 
-MODULE_DESCRIPTION("SUN6I 10/100/1000Mbps Ethernet device driver");
+MODULE_DESCRIPTION("SUN6I 10/100/1000Mbps Ethernet device driver (pcDuino3Nano Mods)");
 MODULE_AUTHOR("Shuge <shugeLinux@gmail.com>");
 MODULE_LICENSE("GPL v2");
